dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/legacy/CountTelnetHandlerTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.qos.legacy;
+
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.qos.legacy.channel.MockChannel;
+import org.apache.dubbo.qos.legacy.service.DemoService;
+import org.apache.dubbo.remoting.telnet.TelnetHandler;
+import org.apache.dubbo.remoting.telnet.support.TelnetUtils;
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.RpcStatus;
+import org.apache.dubbo.rpc.protocol.dubbo.DubboProtocol;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+
+public class CountTelnetHandlerTest {
+
+    private TelnetHandler handler = new CountTelnetHandler();
+    private MockChannel mockChannel;
+    private Invoker<DemoService> mockInvoker;
+    private URL url = URL.valueOf("dubbo://127.0.0.1:20884/demo");
+
+    @BeforeEach
+    public void setUp() {
+        mockChannel = new MockChannel(url);
+        mockInvoker = mock(Invoker.class);
+        given(mockInvoker.getInterface()).willReturn(DemoService.class);
+        given(mockInvoker.getUrl()).willReturn(url);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        ProtocolUtils.closeAll();
+        mockChannel.close();
+        reset(mockInvoker);
+    }
+
+    @Test
+    public void test() throws Exception {
+        String methodName = "sayHello";
+        String message = "org.apache.dubbo.qos.legacy.service.DemoService sayHello 1";
+
+        DubboProtocol.getDubboProtocol().export(mockInvoker);
+        RpcStatus.beginCount(url, methodName);
+        RpcStatus.endCount(url, methodName, 10L, true);
+        handler.telnet(mockChannel, message);
+
+        Thread.sleep(1000);
```

Comment:
Try use CountDownLatch to wait async computing. Using timeout without any check is unreliable.

Reply:
Done